src/lib.rs

```diff
@@ -523,26 +519,28 @@ impl Jieba {
             }
             buf_indices.clear();
         }
-        words
     }
 
     fn cut_internal<'a>(&self, sentence: &'a str, cut_all: bool, hmm: bool) -> Vec<&'a str> {
-        let mut words = Vec::new();
+        let heuristic_capacity = sentence.chars().count() / 2;
+        let mut words = Vec::with_capacity(heuristic_capacity);
         let re_han: &Regex = if cut_all { &*RE_HAN_CUT_ALL } else { &*RE_HAN_DEFAULT };
         let re_skip: &Regex = if cut_all { &*RE_SKIP_CUT_ALL } else { &*RE_SKIP_DEAFULT };
         let splitter = SplitMatches::new(&re_han, sentence);
+        let mut buf_indices = Vec::with_capacity(heuristic_capacity);
```

Comment:
I don't think we need to pass this around? We can allocate this in `cut_dag_hmm` & `cut_dag_no_hmm`.